x-pack/plugins/alerting/common/rule.ts

```diff
@@ -92,7 +92,7 @@ export interface RuleAggregations {
 }
 
 export interface RuleLastRun {
-  outcome: RuleLastRunOutcomes;
+  outcome: RuleLastRunOutcomes | null;
```

Comment:
Need to have `outcome` initialized to null so we can set an intermediate outcome in our security rule type wrapper. Can be overwritten in `x-pack/plugins/alerting/server/lib/last_run_status.ts`, which runs after rule execution has ended.